eth_utils/abi.py

```diff
@@ -42,27 +83,682 @@ def collapse_if_tuple(abi: Dict[str, Any]) -> str:
+    """
+    Return names for each output from the function ABI.
+
+    :param function_abi: Function ABI.
+    :type function_abi: `ABIFunction`
+    :return: Names for each function output in the function ABI.
+    :rtype: `List[str]`
+
+    .. doctest::
+
+        >>> from eth_utils import get_abi_output_names
+        >>> abi = {
+        ...   'constant': False,
+        ...   'inputs': [
+        ...     {
+        ...       'name': 's',
+        ...       'type': 'uint256'
+        ...     }
+        ...   ],
+        ...   'name': 'f',
+        ...   'outputs': [
+        ...     {
+        ...       'name': 'name',
+        ...       'type': 'string'
+        ...     },
+        ...     {
+        ...       'name': 's',
+        ...       'type': 'uint256'
+        ...     }
+        ...   ],
+        ...   'payable': False,
+        ...   'stateMutability': 'nonpayable',
+        ...   'type': 'function'
+        ... }
+        >>> get_abi_output_names(abi)
+        ['name', 's']
+    """
+    if "outputs" not in function_abi or function_abi["type"] != "function":
+        raise ValueError(
+            f"Outputs only supported for ABI type 'function'. Provided"
+            f" ABI type was '{function_abi['type']}'."
+        )
+    return [arg["name"] for arg in function_abi["outputs"]]
+
+
+def get_abi_output_types(function_abi: ABIFunction) -> List[str]:
+    """
+    Return types for each output from the function ABI.
+
+    :param function_abi: Function ABI.
+    :type function_abi: `ABIFunction`
+    :return: Types for each function output in the function ABI.
+    :rtype: `List[str]`
+
+    .. doctest::
 
+        >>> from eth_utils import get_abi_output_types
+        >>> abi = {
+        ...   'constant': False,
+        ...   'inputs': [
+        ...     {
+        ...       'name': 's',
+        ...       'type': 'uint256'
+        ...     }
+        ...   ],
+        ...   'name': 'f',
+        ...   'outputs': [
+        ...     {
+        ...       'name': 'name',
+        ...       'type': 'string'
+        ...     },
+        ...     {
+        ...       'name': 's',
+        ...       'type': 'uint256'
+        ...     }
+        ...   ],
+        ...   'payable': False,
+        ...   'stateMutability': 'nonpayable',
+        ...   'type': 'function'
+        ... }
+        >>> get_abi_output_types(abi)
+        ['string', 'uint256']
 
-def function_abi_to_4byte_selector(function_abi: Dict[str, Any]) -> bytes:
-    function_signature = _abi_to_signature(function_abi)
+    """
+    if function_abi["type"] == "function":
+        return [get_normalized_abi_arg_type(arg) for arg in function_abi["outputs"]]
+
+    raise ValueError(
+        f"Outputs only supported for ABI type 'function'. Provided"
+        f" ABI type was '{function_abi['type']}'."
+    )
+
+
+def function_signature_to_4byte_selector(function_signature: str) -> bytes:
+    r"""
+    Return the 4-byte function signature from a function signature string.
```

Comment:
[source](https://docs.soliditylang.org/en/latest/abi-spec.html#function-selector)

```suggestion
    Return the 4-byte function selector from a function signature string.
```